examples/learning/learn_associations.ipynb

```diff
@@ -0,0 +1,330 @@
+      "\n",
+      "Voja's rule works by moving the encoders of the active neurons toward the current input. This can be stated succinctly as,\n",
+      "\n",
+      "$$\n",
+      "\\Delta e_i = \\kappa a_i (x - e_i)\n",
+      "$$\n",
+      "\n",
+      "where $e_i$ is the encoder of the $i^{th}$ neuron, $\\kappa$ is a modulatory learning rate (positive to move towards, and negative to move away), $a_i$ is the filtered activity of the $i^{th}$ neuron, and $x$ is the input vector encoded by each neuron. To see how this is related to Oja's rule, substituting $e_i$ with the row of weights $W_i$, $x$ for the pre-synaptic activity vector $b$, and letting $s = 1 / a_i$ be a dynamic normalizing factor, gives\n",
+      "\n",
+      "$$\n",
+      "\\Delta W_i = \\kappa a_i (b - s a_i W_i)\n",
+      "$$\n",
+      "\n",
+      "which is the update rule for a single row using Oja. For more details, see [Learning large-scale heteroassociative memories in spiking neurons](http://compneuro.uwaterloo.ca/publications/voelker2014a.html).\n",
+      "\n",
+      "This notebook will lead the reader through a basic example of building a network that can store and recall new associations."
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 1: Configure some example data\n",
+      "\n",
+      "First, we will setup some keys (inputs) and values (outputs) for our network to store and recall."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "import numpy as np\n",
+      "import matplotlib.pyplot as plt\n",
+      "%matplotlib inline\n",
+      "\n",
+      "import nengo"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [],
+     "prompt_number": null
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "num_items = 5\n",
+      "\n",
+      "d_key = 2\n",
+      "d_value = 4\n",
+      "\n",
+      "np.random.seed(7)\n",
+      "keys = nengo.dists.UniformHypersphere(surface=True).sample(num_items, d_key)\n",
+      "values = nengo.dists.UniformHypersphere(surface=False).sample(num_items, d_value)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [],
+     "prompt_number": null
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "An important quantity is the largest dot-product between all pairs of keys, since a neuron's intercept should not go below this value if it's positioned between these two keys. Otherwise, the neuron will move back and forth between encoding those two inputs."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "intercept = (np.dot(keys, keys.T) - np.eye(num_items)).flatten().max()\n",
+      "print (\"Intercept: %s\" % intercept)"
```

Comment:
Minor nitpick: no space between `print` and `(` (should look like a function for Py3).